daemon/daemon.go

```diff
@@ -1474,42 +1476,38 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState) (err err
 				reason = "Denied by policy"
 			}
 
+			var epAddr string     // the address of the endpoint that originated the request
+			var serverAddr string // the address of the DNS target
 			var ingress = msg.Response
 			var flowType accesslog.FlowType
 			if ingress {
 				flowType = accesslog.TypeResponse
+				epAddr = dstAddr
+				serverAddr = srcAddr
 			} else {
 				flowType = accesslog.TypeRequest
+				epAddr = srcAddr
+				serverAddr = dstAddr
 			}
 
-			var ep *endpoint.Endpoint
-			srcID := uint32(0)
-			srcIP, _, err := net.SplitHostPort(srcAddr)
+			var serverPort int
+			_, serverPortStr, err := net.SplitHostPort(serverAddr)
 			if err != nil {
-				// dstAddr is used instead, below
 				log.WithError(err).Error("cannot extract endpoint IP from DNS request")
 			} else {
-				ep = endpointmanager.LookupIPv4(srcIP)
-				if ep != nil {
-					srcID = ep.GetIdentity().Uint32()
+				if serverPort, err = strconv.Atoi(serverPortStr); err != nil {
+					log.WithError(err).WithField(logfields.Port, serverPortStr).Error("cannot parse destination port")
```

Comment:
I didn't return an error because serverPort being wrong (and 0) only affected the metrics. It seemed back to drop the DNS response just for that (in contrast, when ep == nil we can't report any events, but it also means something is really wrong for us, and we shouldn't send a packet along to something that isn't an endpoint).